common/src/main/java/org/apache/seata/common/util/ReflectionUtil.java

```diff
@@ -705,6 +705,45 @@ public static Map<String, Object> getAnnotationValues(Annotation annotation) thr
         return getFieldValue(h, "memberValues");
     }
 
+    /**
+     *
+     * @param targetClass the target class
+     * @param methodName the method name
+     * @param paramTypes the param types
+     * @param annotationClass the annotation class
+     * @return the annotation
+     * @param <A>
+     */
+    public static <A extends Annotation> A findAnnotationInHierarchy(
```

Comment:
I believe method-level annotation data should take precedence. Only if a method does not have that annotation should we check the corresponding method in the superclass. This ensures that if the same annotation appears with different configured values, the value on the method is the one that ultimately applies.